src/compiler/utilities.ts

```diff
@@ -903,7 +903,7 @@ namespace ts {
 
     export function isConst(node: Node): boolean {

```

Comment:
for other reviews: all the changes below here are just refactorings based on an improved definition of walkupBindingElementsAndPatterns.